autopilot/contractor.go

```diff
@@ -259,10 +266,28 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) (
 		}
 	}
 
-	// run renewals
+	// split toRenew up in contracts that are already in the set and the ones
+	// that aren't.
+	// TODO: same for refresh
+	var renewedInSet, renewedNotInSet []contractInfo
```

Comment:
I think we don't need all of this. What if we do the following:

```go
// 1. we keep runContractChecks exactly the same
// 2. we update "calculate 'limit' amount of contracts we want to renew" and add the following:

for _, c := range toRenew {
    limit++
}
for len(updatedSet)+limit < int(state.cfg.Contracts.Amount) && limit < len(toRenew) {
    // as long as we're missing contracts, increase the renewal limit
    limit++
}

// 3. we add a similar limit for `toRefresh`
```

That way I think you accomplish exactly what you want and you make the change more in-line with the code that we have currently. Less "exceptions to the rule" and just using `limit` we have already.